### PR TITLE
Avoid crashes that might occur while initializing map providers

### LINKED ISSUE #3846: Broadcast of Intent { act=com.mapbox.scheduler_flusher flg=0x14 (has extras) }
#### Software and hardware versions 
Collect v1.27.0-beta

#### Problem description
The QA team has been reporting that there are more ANRs in the newest version. I investigated the issue and discovered that there is one new ANR that is the most common one now and and it's probably this.

`Broadcast of Intent { act=com.mapbox.scheduler_flusher flg=0x14 (has extras) }`

https://play.google.com/apps/publish/?account=6923365842552008664&dev_acc=00659255839475882647#AndroidMetricsErrorsPlace:p=org.odk.collect.android&appVersion=3782&errorType=ANR&clusterName=apps/org.odk.collect.android/clusters/17827f0f&detailsAppVersion=3782&detailsSpan=7

I guess it has something to do with [Display mapbox map to perform initialization](https://github.com/getodk/collect/pull/3819/commits/9cede13f5ad9e5f060cdd2941a1a8720903b287b)

I was able to find two similar issues on the internet:
https://github.com/mapbox/mapbox-gl-native-android/issues/263
https://github.com/mapbox/mapbox-events-android/issues/310

in the firs one there is a comment:
> the com.mapbox.scheduler_flusher stems from Telemetry.

in the second one the author claims that adding:
`-keep class com.mapbox.android.telemetry.** { *; }`
fixed the issue but also someone else claims that it shouldn't be needed because it is already fixed in mapbox... maybe the issue is back?

Should we try or you have better ideas @lognaturel  @seadowg?

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #3846 

#### What has been done to verify that this works as intended?
I reviewed implemented changes.

#### Why is this the best possible solution? Were any other approaches considered?
The first commit is not strictly a fix for the issue but should catch possible exceptions like:
https://console.firebase.google.com/u/0/project/api-project-322300403941/crashlytics/app/android:org.odk.collect.android/issues/45c613c546aa3cea42be7d9fa55c286d?time=last-seven-days&sessionId=5EF99A4C0180000114A81ED19C1D1285_DNE_0_v2

The second commit I prepared based on:
https://github.com/mapbox/mapbox-gl-native-android/issues/126#issuecomment-650708237
and
https://docs.mapbox.com/android/maps/overview/#5-lifecycle-methods

@lognaturel you said that we should at least handle on destroy but in bot cases (links) they say that we should handle all lifecycle methods so just in case I think it's a good idea to do that.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
It shouldn't change anything in behavior. It should just prevent from crashes so i'm not even sure if this pr requires testing.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

**Review by lognaturel — APPROVED** [review]
I think we could do without some of the more esoteric lifecycle events like `onLowMemory` but I'm also fine to not risk it. All of these call out to native code so who knows what's going on. Good call on the first commit.

**Comment by lognaturel** [comment]
I agree we can merge this without testing. I'll give others a couple days to comment if desired (@seadowg, @getodk/testers).

File: collect_app/src/main/java/org/odk/collect/android/activities/MainMenuActivity.java [hunk 1]
```diff
@@ -314,6 +316,14 @@ public void onClick(View v) {
         }
     }
 
+    @Override
+    protected void onStart() {
+        super.onStart();
+        if (mapView != null) {
+            mapView.onStart();
```
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/activities/MainMenuActivity.java]: I'm just wondering if it's possible to put the `MapView` in a `Fragment` so we can have all the lifecycle "forwarding" in there. Thinking we have like a `MapBoxInitializationFragment` that the `MainMenuActivity` can host. Then we can keep all this noise out of `MainMenuActivity`.

Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/activities/MainMenuActivity.java]: I think it's a good idea and I did it please take a look.

**Review by lognaturel — APPROVED** [review]

**Comment by lognaturel** [comment]
I'm delighted with how the code has turned out. Thanks, @grzesiek2010!

**Comment by lognaturel** [comment]
@getodk/testers This pulls the mapbox initialization for offline layers out of the `MainMenuActivity`. We've deemed it very low risk and we can't reproduce the crash that it fixes so we decided to do without QA but wanted to let you know it happened in case you think of possible risk we haven't considered.

**Pull request merged**